Makefile.js

```diff
@@ -199,7 +199,7 @@ function generateRuleIndexPage() {
             if (rule.meta.deprecated) {
                 ruleTypesData.deprecated.push({
                     name: basename,
-                    replacedBy: rule.meta.replacedBy || [],
+                    replacedBy: rule.meta.deprecated.replacedBy?.map(replacement => replacement.rule.name) ?? [],
```

Comment:
The build is now failing because the autogenerated rules.json file (`docs/src/_data/rules.json`) contains the names of replacement rules that are in some other plugin, so the link checker fails because it finds links to pages that don't exist like `/docs/head/rules/function-call-spacing`.

If we only want to show replacement rules that are in the core as we are currenly doing then we should filter out rules that aren't in the core by checking that `replacement.plugin` is undefined.

Reply:
I think it would be good to distinguish between core replacement rules and external. It's helpful to have links to the replacement rules in the docs, and if the deprecation information has a URL for that replacement rule, we should use that instead.